### PR TITLE
Complete doc overhaul

### PR DESCRIPTION
**Pull request by lorenzwalthert**
This PR improves documentation in various ways, in particular: 

* README: 
  * moves adoption and examples out to vignettes. 
  * move feature overview to intro vignette.
  * drop contributing guidelines.
  * updates badges (gh actions, travis appveyor etc).
  * add motivation
* big review of intro to styler vignette and rebranding as 'Get started' article with direct link in navbar, in particular:
  * caching
  * stylerignore
  * dry mode
  * scope specification
  * alignment detection
* caching and alignment: better cross-linking and documentation.
* add vignette with examples for `strict = FALSE`.
* add jupyter lab integration to adaption vignette.

### CODE REVIEW

File: vignettes/styler.Rmd [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+# Dry mode
+
+As of version `1.3.2`, styler has a dry mode which avoids writing output to the
+file(s) you want to format. The following options are available:
+
+- *off* (default): Write back to the file if applying styling changes the
+  input.
+
+- *on*: Like *off*, but not writing back.
```
Comment (jonmcalder) [inline vignettes/styler.Rmd:174]:
```suggestion
- *on*: Applies styling and returns the results without writing changes (if any) back to the file(s).
```

File: vignettes/styler.Rmd [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  }
+)
+out
+```
+
+# More configuration options
+
+### Roxygen code example styling
+
+This is enabled by default, you can turn it of with `include_roxygen_examples`.
```
Comment (jonmcalder) [inline vignettes/styler.Rmd:197]:
```suggestion
This is enabled by default, you can turn it off with `include_roxygen_examples = FALSE`.
```

File: vignettes/strict.Rmd [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      },
+      ""
+    )
+  }
+))
+```
+
+This vignette shows how output from styler might differ when `strict = FALSE`. 
+For brevity, we don't show the output of `strict = TRUE`, but it should be 
+pretty simple for the user to derive it form the bullet or simply paste the
```
Comment (jonmcalder) [inline vignettes/strict.Rmd:38]:
```suggestion
pretty simple for the user to derive it from the bullet point(s) or simply paste the 
```

File: vignettes/adoption.Rmd [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* in the *format-all* command for Emacs in 
+  [emacs-format-all-the-code](https://github.com/lassik/emacs-format-all-the-code).
+
+* As a [Jupyterlab code
+  formatter](https://jupyterlab-code-formatter.readthedocs.io/en/latest/installation.html#r-code-formatters).
+
+* for pretty-printing [drake](https://github.com/ropensci/drake) workflow data
+  frames with `drake::drake_plan_source()`.
+
+You know another way to use styler that is not listed here? Please let us know,
```
Comment (jonmcalder) [inline vignettes/adoption.Rmd:55]:
```suggestion
Do you know another way to use styler that is not listed here? Please let us know,
```

File: vignettes/adoption.Rmd [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+by [opening an issue](https://github.com/r-lib/styler/issues), we'll extend the
```
Comment (jonmcalder) [inline vignettes/adoption.Rmd:56]:
```suggestion
by [opening an issue](https://github.com/r-lib/styler/issues) and we'll extend the
```

**Review by jonmcalder — CHANGES_REQUESTED** [review]
I've suggested a few small changes but overall it looks good to me!

**Comment by lorenzwalthert** [comment]
Thanks @jonmcalder 🥳

**Pull request merged**